src/scripting/AbstractScriptManager.java

```diff
@@ -27,58 +27,54 @@
 import java.io.FileReader;
 import java.io.IOException;
 
-import javax.script.Invocable;
-import javax.script.ScriptEngine;
-import javax.script.ScriptEngineManager;
-import javax.script.ScriptException;
+import javax.script.*;
 
 import constants.ServerConstants;
+import jdk.nashorn.api.scripting.NashornScriptEngine;
+import jdk.nashorn.api.scripting.NashornScriptEngineFactory;
 import tools.FilePrinter;
 
 /**
  *
  * @author Matze
  */
 public abstract class AbstractScriptManager {
-
-    protected ScriptEngine engine;
-    private ScriptEngineManager sem;
+    private ScriptEngineFactory sef;
 
     protected AbstractScriptManager() {
-        sem = new ScriptEngineManager();
+        sef = new ScriptEngineManager().getEngineByName("javascript").getFactory();
     }
 
-    protected Invocable getInvocable(String path, MapleClient c) {
+    protected NashornScriptEngine getScriptEngine(String path) {
         path = "scripts/" + path;
-        engine = null;
-        if (c != null) {
-            try {
-                engine = c.getScriptEngine(path);
-            } catch (NullPointerException npe) {
-                c = null;   // player disconnected
+        File scriptFile = new File(path);
+        if (!scriptFile.exists()) {
+            return null;
+        }
+        NashornScriptEngine engine = (NashornScriptEngine) sef.getScriptEngine();
+        try (FileReader fr = new FileReader(scriptFile)) {
+            if (ServerConstants.JAVA_8){
+                engine.eval("load('nashorn:mozilla_compat.js');" + System.lineSeparator());
             }
+            engine.eval(fr);
+        } catch (final ScriptException | IOException t) {
+            FilePrinter.printError(FilePrinter.INVOCABLE + path.substring(12), t, path);
+            return null;
         }
+
+        return engine;
+    }
+
+    protected NashornScriptEngine getScriptEngine(String path, MapleClient c) {
+        String cachePath = "scripts/" + path;
```

Comment:
Here's the issue for scripts not loading.

![image](https://user-images.githubusercontent.com/9257761/61569921-f45f1d80-aa5f-11e9-8b52-51add11da4a0.png)

In the subsequent call for `engine = getScriptEngine(cachePath);`, "scripts/" gets appended on "String path" here and also inside `NashornScriptEngine getScriptEngine(String path)`.